File: aladdinworks2_ui/src/services/RackAssetService.js
import http from "../http-common"; 

class RackAssetService {
  getAllRackAssets(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/rackAsset/rackAssets`, searchDTO);
  }

  get(rackAssetId) {
    return this.getRequest(`/rackAsset/${rackAssetId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/rackAsset?field=${matchData}`, null);
  }

  addRackAsset(data) {
    return http.post("/rackAsset/addRackAsset", data);
  }

  update(data) {
  	return http.post("/rackAsset/updateRackAsset", data);
  }
  
  uploadImage(data,rackAssetId) {
  	return http.postForm("/rackAsset/uploadImage/"+rackAssetId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RackAssetService();

File: src/main/java/com/aladdinworks2/service/RackAssetService.java
package com.aladdinworks2.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks2.domain.RackAsset;
import com.aladdinworks2.dto.RackAssetDTO;
import com.aladdinworks2.dto.RackAssetSearchDTO;
import com.aladdinworks2.dto.RackAssetPageDTO;
import com.aladdinworks2.dto.RackAssetConvertCriteriaDTO;
import com.aladdinworks2.service.GenericService;
import com.aladdinworks2.dto.common.RequestDTO;
import com.aladdinworks2.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RackAssetService extends GenericService<RackAsset, Integer> {

	List<RackAsset> findAll();

	ResultDTO addRackAsset(RackAssetDTO rackAssetDTO, RequestDTO requestDTO);

	ResultDTO updateRackAsset(RackAssetDTO rackAssetDTO, RequestDTO requestDTO);

    Page<RackAsset> getAllRackAssets(Pageable pageable);

    Page<RackAsset> getAllRackAssets(Specification<RackAsset> spec, Pageable pageable);

	ResponseEntity<RackAssetPageDTO> getRackAssets(RackAssetSearchDTO rackAssetSearchDTO);
	
	List<RackAssetDTO> convertRackAssetsToRackAssetDTOs(List<RackAsset> rackAssets, RackAssetConvertCriteriaDTO convertCriteria);

	RackAssetDTO getRackAssetDTOById(Integer rackAssetId);







}





